### DISCUSSION TITLE
SocketIO not working on Tauri-built NextJS desktop app

### DISCUSSION THREAD

**Discussion by WaelMajdoub** [tauri-apps/tauri#6400] [Q&A]
Hi Tauri community,

I'm building a desktop app with Next.js and Tauri, and I'm using Socket.IO for real-time communication between the client and the server. When I run the app with "yarn dev" command, the app works fine with Socket.IO and the socket client. However, when I build the app to a desktop application for Windows, the Socket.IO connection doesn't seem to work anymore.

It's worth noting that the issue only occurs in the desktop version of the app, and not when running the app locally.

Has anyone encountered a similar issue with Socket.IO in a Tauri-built app, and if so, how did you resolve it? I'd appreciate any help or insights on this matter.

Thanks in advance!

**Comment by FabianLars** [comment]
Are you using a http/ws connection? If so then this may be related to https://github.com/tauri-apps/tauri/issues/3007. Could also be a CSP i think.

Anyway, try building in debug mode: `tauri build --debug` (if you use npm it needs an extra `--` like `npm run tauri build -- --debug`) - output in `target/debug/`. This build will have the devtools enabled so you can check its console for errors.

**Reply by raunakdoesdev** [reply]
Any update on this? I am also struggling to use socketio client on the frontend for a tauri app.

**Reply by FabianLars** [reply]
@sauhaardac Still waiting for OP's answers to my questions 😅 Would you mind answering them for your use-case too?

**Reply by raunakdoesdev** [reply]
I actually solved my issue! @FabianLars Socket.IO is working without any problems. What solved the problem was forcing socket.io to use the websocket transport and to replace http// with ws:// in the URL. For whatever reason, it worked without this in the browser but making these changes made the protocol work within Tauri too.
```
    const s = io(
      socketUrl.replace("http://", "ws://").replace("https://", "ws://"),
      { transports: ["websocket"] }
    );
```

**Reply by raunakdoesdev** [reply]
Nevermind, this was all on dev mode. Am getting the following error when running socket.io in build mode with --debug:
```
[Error] WebSocket connection to 'ws://tauri/socket.io/?EIO=4&transport=websocket' failed: A server with the specified hostname could not be found. (x8)
```

**Reply by r3wt** [reply]
@sauhaardac websockets over ssl protocol is `wss://`, not `ws://`.

**Reply by raunakdoesdev** [reply]
Yep fixed that, thanks. Turns out you can operate without SSL on macOS but it's needed for windows.

**Reply by RHAM91** [reply]
I found a solution, it is the following:
in the tauri.conf.json file go to the "security" key you should add this:

```
 "security": {
      "dangerousUseHttpScheme": true,
      "csp": null
  }
```

this will allow mixed content when we try to fetch http endpoints.

I hope I've been useful

**Reply by PMLS3** [reply]
dangerousUseHttpScheme is not an option anymore, is there another fix?

**Reply by FabianLars** [reply]
In v1 this is still an option and in v2 this is now the default so the option was removed (we'll probably add it back at some point)

**Comment by prabhatm102** [comment]
> [!IMPORTANT]
> fdhfdhdf

hrt
h
rt
<details><summary>Details</summary>
<p>

mhmhgmhgmghmghmghmhmg

</p>
</details>
